devito/finite_differences/coefficients.py

```diff
@@ -224,6 +224,33 @@ def generate_subs(i):
         rules = {}
         for i in self.coefficients:
             rules.update(generate_subs(i))
+        return rules
+
+    def update_rules(self, obj):
```

Comment:
I don't think this process should be happening here. This is the job of the `Coefficient` object. This is where any staggering etc should be being taken into account and hence allowing this logic to be abstracted away from the `Substitutions` object.